gu-hub-api.yaml

```diff
@@ -0,0 +1,582 @@
+swagger: '2.0'
+info:
+  title: Golem unlimited low level hub API
+  description: API description in Markdown.
+  version: 1.0.0
+host: '127.0.0.1:61622'
+tags:
+  - name: hub-session
+    description: HUB Session managment.
+  - name: peer-info
+    description: Unlimited network peer info managment.
+  - name: peer-session
+  - name: lob-man
+    description: Large objects managment
+
+schemes:
+  - http
+security:
+  - serviceToken: []
+  - systemName: []
+paths:
+  /peer:
+    get:
+      tags:
+        - peer-info
+      operationId: listPeers
+      summary: Returns a list hub peers.
+      parameters:
+        - $ref: '#/parameters/offset'
+        - $ref: '#/parameters/limit'
+      produces:
+        - application/json
+      responses:
+        '200':
+          description: OK
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/PeerInfo'
+  /peer/{nodeId}:
+    parameters:
+      - $ref: '#/parameters/nodeId'
+    get:
+      tags:
+        - peer-info
+      operationId: getPeerDetails
+      summary: 'Returns detailed peer info'
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/PeerInfo'
+
+  /sessions:
+    get:
+      tags:
+        - hub-session
+      summary: Lists current hub sessions.
+      operationId: listHubSessions
+      parameters:
+        - name: limit
+          type: integer
+          in: query
+          allowEmptyValue: true
+          default: 50
+        - name: offset
+          type: integer
+          in: query
+          allowEmptyValue: true
+          default: 0
+      responses:
+        '200':
+          description: OK list sessions
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/HubSession'
+    post:
+      tags:
+        - hub-session
+      operationId: createHubSession
+      parameters:
+        - name: newSessionSpec
+          in: body
+          schema:
+            $ref: '#/definitions/HubSession'
+      responses:
+        '201':
+          description: Session created
+          schema:
+            type: string
+            description: Created hub session id
+        '500':
+          description: Error
+          schema:
+            type: string
+            description: Error message
+  '/sessions/{sessionId}':
```

Comment:
once url is with plural otherwise in singlar. it is not consistent.

https://hackernoon.com/restful-api-designing-guidelines-the-best-practices-60e1d954e7c9

"Conclusion: The paths should contain the plural form of resources and the HTTP method should define the kind of action to be performed on the resource."

Reply:
There are some exceptions not mentioned in this article, e.g.,
- **`cancel`** action/verb being part of an URL ([real life example](https://cloud.google.com/bigquery/docs/reference/rest/v2/#Jobs)), because the `DELETE` method, even if it is not used to delete resource of this kind, would confuse users if used to cancel
- **`serviceAccount `** singular within an URL ([real life example](https://cloud.google.com/bigquery/docs/reference/rest/v2/#Projects)), because there is just single resource with this name under this URL. This API has for example `config` which will fall into same exception.